src/layers/image-label-layer.ts

```diff
@@ -0,0 +1,103 @@
+import { MultiscaleImageLayer, getImageSize } from "@hms-dbmi/viv";
+import { TileLayer } from "deck.gl";
+import { clamp } from "math.gl";
+import * as utils from "../utils";
+
+import type { PixelData } from "@vivjs/types";
+import { BitmapLayer } from "deck.gl";
+import type { Matrix4 } from "math.gl";
+import type { ZarrPixelSource } from "../ZarrPixelSource";
+
+export type ImageLabelLayerProps = {
+  id: string;
+  loader: Array<ZarrPixelSource>;
+  selection: Array<number>;
+  opacity: number;
+  visible?: boolean;
+  modelMatrix?: Matrix4;
+  color?: [r: number, g: number, b: number];
+};
+
+// just subclass viv for now
+export default class _ImageLabelLayer extends MultiscaleImageLayer<string[]> {
+  constructor(props: ImageLabelLayerProps) {
+    super({
+      id: `labels-${props.id}`,
+      loader: props.loader,
+      opacity: props.opacity,
+      // @ts-expect-error - Viv types aren't great
+      selections: [props.selection],
+      colors: [props.color ?? [255, 255, 255]],
```

Comment:
> I know that we will at least need to fine a way to make `ImageLabelLayer` have a transparent background.

I noticed that viv provides a `ColorPaletteExtension` class. I tried to apply it here and it seems to produce the transparent background. Could that be a solution or you need something different?

```js
useTransparentColor: true,
transparentColor: [0, 0, 0],
extensions: [new ColorPaletteExtension()],
```

Reply:
I forgot about this 😅 , thank you!